### PR TITLE
Misp, closes #1955

### LINKED ISSUE #1955: MISP observable analyzer adjustments
We should revision that analyzer:

- [ ] "timeout" in PyMISP should be configurable by the user
- [ ] new option in the search: "published" to get only published event
- [ ] new option in the search: "metadata" to have lighter queries but less data

Reference:
https://github.com/MISP/PyMISP/blob/main/docs/tutorial/Search-FullOverview.ipynb

### PR DESCRIPTION
**Pull request by g4ze**
closes #1955 

# Description

add new parameters
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [x] The pull request is for the branch `develop`
- [x] A new plugin (analyzer, connector, visualizer, playbook, pivot or ingestor) was added or changed, in which case:
    - [x] I strictly followed the documentation ["How to create a Plugin"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-add-a-new-plugin)
    - [x] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Advanced-Usage.md) was updated (in case the plugin provides additional optional configuration).
    - [ ] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [ ] I have dumped the configuration from Django Admin using the `dumpplugin` command and added it in the project as a data migration. (["How to share a plugin with the community"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-share-your-plugin-with-the-community))
    - [ ] If a File analyzer was added and it supports a mimetype which is not already supported, you added a sample of that type inside the archive `test_files.zip` and you added the default tests for that mimetype in [test_classes.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_classes.py).
    - [ ] If you created a new analyzer and it is free (does not require API keys), please add it in the `FREE_TO_USE_ANALYZERS` playbook by following [this guide](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-modify-a-plugin).
    - [ ] Check if it could make sense to add that analyzer/connector to other [freely available playbooks](https://intelowl.readthedocs.io/en/develop/Usage.html#list-of-pre-built-playbooks).
    - [ ] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results.
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] I have added tests for the feature/bug I solved (see `tests` folder). All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Contribute.md)).
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.

### Important Rules
- If you miss to compile the Checklist properly, your PR won't be reviewed by the maintainers.
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review. After being reviewed and received a "change request", you should explicitly ask for a review again once you have made the requested changes.

### CODE REVIEW

**Comment by g4ze** [comment]
hey! @mlodic I had a bit of struggle in setting up MISP, now everything works fine (hopefully). Now that we new vaiables are incorporated,
```
intelowl_celery_worker_default  | Something went wrong (400): {'name': 'Restsearch queries using GET and no parameters are not allowed. If you have passed parameters via a JSON body, make sure you use POST requests.', 'message': 'Restsearch queries using GET and no parameters are not allowed. If you have passed parameters via a JSON body, make sure you use POST requests.', 'url': '/events/restSearch'}
```
This is the log im getting, in job report as well. Is it something to do with my MISP?
config:
```
debug:false
limit:50
timeout:30
metadata:false
from_days:90
published:false
ssl_check:false
strict_search:true
filter_on_type:true
enforce_warninglist:true
self_signed_certificate:false
```

File: api_app/analyzers_manager/migrations/0067_update_misp.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+def reverse_migrate(apps, schema_editor):
+    PythonModule = apps.get_model("api_app", "PythonModule")
+
+    pm = PythonModule.objects.get(
+        module="misp.MISP", base_path="api_app.analyzers_manager.observable_analyzers"
+    )
+    pm.analyzerconfig.all().delete()
+    pm.parameter.all().delete()
+    pm.delete()
```
Comment (0ssigeno) [inline api_app/analyzers_manager/migrations/0067_update_misp.py:86]: This `reverse_migration` function it is not actually the reverse of the forward migration

File: api_app/analyzers_manager/observable_analyzers/misp.py [hunk 2]
```diff
@@ -79,6 +85,11 @@ def run(self):
                     f"Observable {self.observable_classification} not supported."
                     "Currently supported are: ip, domain, hash, url, generic."
                 )
+        if self.published:
+            params["published"] = True
+        if self.metadata:
+            params["metadata"] = True
```
Comment (0ssigeno) [inline api_app/analyzers_manager/observable_analyzers/misp.py:91]: Both ifs are not necessary:
```python3
params["published"] = self.published
params["metada"] = self.metadata
```

**Review by 0ssigeno — CHANGES_REQUESTED** [review]

File: api_app/analyzers_manager/migrations/0067_update_misp.py (hunk 1 above)
Comment (g4ze) [reply api_app/analyzers_manager/migrations/0067_update_misp.py]: okay so is there a way to verify what exactly the reverse migration function does? 
i tried applying and unapplying the migration to verify the functionality in the latest commit.

**Review by 0ssigeno — APPROVED** [review]

File: api_app/analyzers_manager/migrations/0067_update_misp.py (hunk 1 above)
Comment (0ssigeno) [reply api_app/analyzers_manager/migrations/0067_update_misp.py]: What you described it is the best way. 
Apply the migration -> check the state -> reverse the migration -> check the state

If the question was about if there is some automatically check to verify these steps, the answer unfortunately it is not.
Or at least I do not know a way, we could maybe implement something but I think it is overkill at the moment

Comment (g4ze) [reply api_app/analyzers_manager/migrations/0067_update_misp.py]: makes sense, understood

**Review by mlodic — APPROVED** [review]

**Pull request merged**